frr-reloader/Dockerfile

```diff
@@ -0,0 +1,6 @@
+FROM frrouting/frr:v7.5.1
+WORKDIR /
+RUN mkdir /var/log/frr
+COPY frr-reloader.sh .
+
+ENTRYPOINT ["/frr-reloader.sh"]
```

Comment:
Could you make use of `frr-reload` from the frr container via a shared volume?  I'd rather not have to worry about keeping a custom image up to date with changes that happen upstream in frr.  It looks like the `frr-reload.py` script is only using Python standard libraries, so it doesn't seem too bad.

Reply:
I believe it's possible, although I did choose that image because vtysh is included there. In a way sharing vtysh and the reload script through a volume would hurt the image's portability(?), since it'll rely on specific mounts and would be unusable on its own, but I'm fine with that if that's the direction you rather.

Reply:
I believe what @russellb  is suggesting is to ship frr-reloader in the speaker image and use it directly as a daemon in the frr container, but it could be tricky because it can't be used as the entry point (since the other container must have the time to move the file to a shared container). I don't see a straightforward way to achieve that, if not by sharing a volume and do some polling to check if the speaker moved the file in the shared volume and then launch the process.
But maybe there's a more straightforward way to do that.

Reply:
Copying what we discussed offline: I'd keep a separate container in the pod for the loader, but based on the frr base image instead of the custom one.
In this way we don't mess up the entry point with multiple processes, and log wise the container logs only the reloader process, but we still don't have to maintain a separate image. WDTY @russellb ?

Reply:
@fedepaol just using initContainers to move the script to a volume that you then mount on frr container would do the trick.

Reply:
@champtar yes, moving the files is straightforward, but having two process in the same container is a bit less. I am a bit more inclined to have two contaiers using the same frr image, one for frr and the other with the reloader.

Reply:
*straightforward after thinking about it I mean, I did not think about it in the first place :P

Reply:
Having 2 containers using the same frr is cheap so go for it

Reply:
done, removed the need for a custom image

Reply:
Also to @russellb 's point (I think this is the point). Could we just share the FRR binaries, sockets, etc from the FRR container to the speaker container via a shared volume and then the speaker could call frr_reload.py directly?

Reply:
> Also to @russellb 's point (I think this is the point). Could we just share the FRR binaries, sockets, etc from the FRR container to the speaker container via a shared volume and then the speaker could call frr_reload.py directly?

Doensn't it feel a bit too hacky? Having a proxy in the frr container (or in a copy of it) that performs the reload won't cause any dependency headaches, but we can't guarantee the opposite.

Reply:
Yep, I am afraid that for this purpouse we need a better interface

Reply:
the less coupling we have between the Metallb speaker and FRR is the better, also we are assuming single FRR instance/speaker is that long term accurate statement ? have we explore using something like ZMQ with configuration in memory and avoid using disk space ?